### DISCUSSION TITLE
QQ官方API的适配器如何获取发消息的人的QQ号？

### DISCUSSION THREAD

**Discussion by the-cold-heart** [nonebot/discussions#520] [问答] [answered]
我尝试使用`event.get_user_id`的方式获取用户的QQ号，但是没有成功，我将`event.get_user_id`的返回值转换成`str`之后发送出来，获得了如下回复，我从中没有找到疑似QQ号的内容
&lt;bound method C2CMessageCreateEvent.get_user_id of C2CMessageCreateEvent(id='ROBOT1.0_hZyxWD3yFUPUHkYxE7-rnMLeM0EeeTYbJ73wnPipYgzhdwxVv94nYYQ-imLn9fC.aQbxG9ktxPL6SdGXH2PqsQ!!', content='获取id', timestamp='2024-09-29T11:18:01+08:00', attachments=None, to_me=True, author=FriendAuthor(id='18AAF2B4EFDC68955DA802EDCB0E683A', user_openid='18AAF2B4EFDC68955DA802EDCB0E683A'), _message=[Text(type='text', data={'text': '获取id'})])&gt;

**Comment by yanyongyu** [comment, marked as answer]
QQ 开放平台的所有应用均无法获取用户的 QQ 号，只能获取到 open_id 和 union_id。请参考官方文档。

**Reply by the-cold-heart** [reply]
这样的话，该如何辨别普通用户和管理员呢？

**Reply by the-cold-heart** [reply]
哦哦，原来open_id也能用来区分啊，谢谢大佬指导

**Reply by Shanshui2024** [reply]
> 这样的话，该如何辨别普通用户和管理员呢？

就没有办法 官方给出的消息体里没有管理员之类的 群聊更不会往机器人侧推送role或者相关的身份信息

这是QQ官方接口的限制 没招（
